userspace/libsinsp/parsers.cpp

```diff
@@ -4921,6 +4921,37 @@ void sinsp_parser::parse_rw_exit(sinsp_evt *evt)
 					(*it)->on_read(evt, data, datalen);
 				}
 			}
+
+			//
+			// Check if recvmsg contains ancillary data. If so, we check for SCM_RIGHTS,
+			// which is used to pass FDs between processes, and update the sinsp state 
+			// accordingly via procfs scan.
+			//
+#ifndef _WIN32
+			if(etype == PPME_SOCKET_RECVMSG_X && evt->get_num_params() >= 5)
+			{
+				parinfo = evt->get_param(4);
+				if(parinfo->m_len > sizeof(struct cmsghdr))
+				{
+					struct cmsghdr *cmsg = (struct cmsghdr *)parinfo->m_val;
+					if(cmsg->cmsg_type == SCM_RIGHTS)
+					{
+						auto scap_tinfo = scap_proc_alloc(m_inspector->m_h);
```

Comment:
i think we need a NULL check here on `scap_tinfo`, `scap_proc_alloc` could return a NULL pointer